docs/custom_proxy.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Custom Proxy for Statsig API
+---
+
+Instead of sending API requests directly to Statsig, you can set up your own environment that proxies requests from your custom domain name to Statsig.
+
+There are many ways to set up custom proxies. We are showing instructions for a few common service providers here.
+
+## AWS CloudFront
+
+### Prerequisites
+
+- Write access to your DNS settings.
+- Write access on your AWS CloudFront console.
+- Access to a SSL certificate for your custom domain.
+
+### Setup
+
+On your [AWS CloudFront console](https://console.aws.amazon.com/cloudfront/),
+
+- Click on Create distribution.
+- In the Origin section,
+
+  - Set the Origin Domain to `api.statsig.com`.
+  - Set the Protocol to `HTTPS only`.
+
+![Origin section](https://user-images.githubusercontent.com/7304774/178337858-834c6762-15b4-410d-91bb-68e04932523e.png)
+
+- In the Default cache behavior section,
+
+  - Set Viewer protocol policy to `Redirect HTTP to HTTPS`.
+  - Set Allowed HTTP methods to `GET, HEAD, OPTIONS, PUT, POST, PATCH, DELETE`.
+
+![Default cache behavior section](https://user-images.githubusercontent.com/7304774/178337879-d2a16abe-46be-4f79-a708-de1a55cae5b1.png)
+
+- In Settings,
+
+  - Add an Alternate domain name (CNAME) to be your preferred domain name to use for the custom proxy, e.g. `statsig.example.com`.
+  - Add a Custom SSL certificate. You will need to follow the [AWS guide for Alternate domain name](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html#alternate-domain-names-requirements) to add a SSL certificate.
+  - Click on Create distribution.
+
+![Settings](https://user-images.githubusercontent.com/7304774/178337890-828e9f37-dd28-43a3-adc1-061052916045.png)
+
+- You will get a Distribution domain name (e.g. `d111111abcdef8.cloudfront.net`) once it is provisioned.
+
+In your DNS settings (depending on your DNS provider),
+
+- Add a CNAME record in your custom DNS record:
+  
+  - Host name: `statsig.example.com`
+  - Type: `CNAME`
+  - Data: `d111111abcdef8.cloudfront.net` (The Distribution domain name from AWS)
+
+### Update Statsig SDK to use your custom domain
+
+When using Statsig SDK, make sure to update the `api` parameter in the [Statsig Options](/client/jsClientSDK#statsig-options) to your custom domain with `/v1/` appended, e.g. `https://statsig.example.com/v1/`.
```

Comment:
Oh interesting, does the proxy forward the entire path?  Is this necessary?

Reply:
yes. It forwards the entire path.